ci/fireci/fireci/uploader.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import requests
+
+from fireci import prow_utils
+
+_logger = logging.getLogger('fireci.uploader')
+
+
+def post_report(
+    test_report,
+    metrics_service_url=os.getenv('METRICS_SERVICE_URL'),
+    access_token=prow_utils.gcloud_identity_token()):
+  """Post a report to the metrics service backend."""
+
+  endpoint = _construct_request_endpoint()
+  headers = {'Authorization': f'Bearer {access_token}', 'Content-Type': 'application/json'}
+  data = json.dumps(test_report)
+
+  _logger.info('Posting to the metrics service ...')
+  _logger.info(f'Request endpoint: {endpoint}')
+  _logger.info(f'Request data: {data}')
+
+  request_url = f'{metrics_service_url}{endpoint}'
+  result = requests.post(request_url, data=data, headers=headers)
+
+  _logger.info(f'Response: {result.text}')
+
+
+def _construct_request_endpoint():
+  repo_owner = os.getenv('REPO_OWNER')
+  repo_name = os.getenv('REPO_NAME')
+  branch = os.getenv('PULL_BASE_REF')
+  base_commit = os.getenv('PULL_BASE_SHA')
+  head_commit = os.getenv('PULL_PULL_SHA')
```

Comment:
PULL_PULL_SHA  : Typo?

Reply:
This env var is actually indeed called `PULL_PULL_SHA`. See prow doc [here](https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables).